azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchImpl.java

```diff
@@ -0,0 +1,414 @@
+
+  @Inject
+  public ContainerizedDispatchImpl(final Props azkProps, final ExecutorLoader executorLoader,
+      final CommonMetrics commonMetrics, final ExecutorApiGateway apiGateway,
+      final ContainerizedImpl containerizedImpl) throws ExecutorManagerException{
+    super(azkProps, executorLoader, commonMetrics, apiGateway);
+    rateLimiter =
+        RateLimiter.create(azkProps.getInt(ContainerizedExecutionManagerProperties.CONTAINERIZED_CREATION_RATE_LIMIT, 20));
+    this.containerizedImpl = containerizedImpl;
+  }
+
+  /**
+   * Get execution ids of all running (unfinished) flows from database.
+   */
+  public List<Integer> getRunningFlowIds() {
+    final List<Integer> allIds = new ArrayList<>();
+    try {
+      getExecutionIdsHelper(allIds, this.executorLoader.fetchUnfinishedFlows().values());
+    } catch (final ExecutorManagerException e) {
+      this.logger.error("Failed to get running flow ids.", e);
+    }
+    return allIds;
+  }
+
+  /**
+   * Get queued flow ids from database. The status for queued flows is READY for
+   * containerization.
+   * @return
+   */
+  public List<Integer> getQueuedFlowIds() {
+    final List<Integer> allIds = new ArrayList<>();
+    try {
+      getExecutionIdsHelper(allIds, this.executorLoader.fetchQueuedFlows(Status.READY));
+    } catch (final ExecutorManagerException e) {
+      this.logger.error("Failed to get queued flow ids.", e);
+    }
+    return allIds;
+  }
+
+  /**
+   * Get size of queued flows. The status for queued flows is READY for containerization.
+   * @return
+   */
+  @Override
+  public long getQueuedFlowSize() {
+    return getQueuedFlowIds().size();
+  }
+
+  /**
+   * Get dispatch method enum for this class.
+   * @return
+   */
+  @Override
+  public DispatchMethod getDispatchMethod() {
+    return DispatchMethod.CONTAINERIZED;
+  }
+
+  /**
+   * This method is used to start queue processor thread. The queue processor thread will pick
+   * execution from queue maintained in database and create a container for it.
+   */
+  @Override
+  public void start() {
+    this.queueProcessor = setupQueueProcessor();
+    this.queueProcessor.start();
+  }
+
+  private QueueProcessorThread setupQueueProcessor() {
+    return new QueueProcessorThread(
+        this.azkProps, this.executorLoader);
+  }
+
+  /**
+   * Get the status for executions maintained in queue. For other dispatch implementations, the
+   * status for executions in queue is PREPARING. But for containerized dispatch method, the status
+   * will be READY.
+   * @return
+   */
+  @Override
+  public Status getStartStatus() {
+    return Status.READY;
+  }
+
+  /**
+   * This method will shutdown the queue processor thread. It won't pick up executions from queue
+   * for dispatch after this. This method will be called when webserver will shutdown.
+   */
+  @Override
+  public void shutdown() {
+    logger.info("Shutting down queue processor thread for containerized dispatch implementation.");
```

Comment:
nit : this.logger

Reply:
As logger is static, I have moved all this.logger to logger.